jit/v2/JIT_v2.0.0.yaml

```
@@ -940,6 +936,263 @@ components:
+      in: query
+      name: portCallServiceID
+      description: |
+        The **Port Call Service Identifier** to filter by. Specifying this filter will (at most) return a single **Port Call Service** object including the `portCallServiceID` (if the `portCallServiceID` exists).
+      schema:
+        type: string
+        format: uuid
+        example: 0342254a-5927-4856-b9c9-aa12e7c00563
+
+    # Timestamp query parameters
+
+    startDateTimeQueryParam:
+      in: query
+      name: startDateTime
+      description: |
+        The **Start Date Time** of the period for which **Port Call Service** information is requested. If a date and time of any **Timestamp** inside a **Port Call Service** matches a date and time on or after (≥) the `startDateTime`, the **Port Call Service** will be included in the result. All matching is done towards local Date and time at the place of the **Port Call Service**. The matched **Timestamp** in the timestamps of the **Port Call Service** must be included in the **Port Call Service** objects returned.
+        If this filter is not provided the default value is `3 months` prior to request time.  The value is populated in [ISO 8601](https://www.iso.org/iso-8601-date-and-time-format.html) date format.
+      schema:
+        type: string
+        format: date-time
+        example: '2020-04-06T07:41:00+08:30'
+    endDateTimeQueryParam:
+      in: query
+      name: endDateTime
+      description: |
+        The **End Date Time** of the period for which **Port Call Service** information is requested. If a date and time of any **Timestamp** inside a **Port Call Service** matches a date and time on or before (≤) the `endDateTime`, the **Port Call Service** will be included in the result. All matching is done towards local Date and time at the place of the **Port Call Service**. The matched **Timestamp** in the timestamps of the **Port Call Service** must be included in the **Port Call Service** objects returned.
+        If this filter is not provided the default value is `6 months` after request time.  The value is populated in [ISO 8601](https://www.iso.org/iso-8601-date-and-time-format.html) date format.
+      schema:
+        type: string
+        format: date-time
+        example: '2020-04-10T07:41:00+08:30'
+
+    # Port Call query parameters
+
+    UNLocationCodeQueryParameter:
+      in: query
+      name: UNLocationCode
+      description: |
+        The **UN Location Code** specifying where a port is located. Specifying this filter will only return **Port Call Service** objects including the `UNLocationCode`.
+      schema:
+        type: string
+        pattern: ^[A-Z]{2}[A-Z2-9]{3}$
+        minLength: 5
+        maxLength: 5
+        example: NLAMS
+    portVisitReferenceQueryParam:
+      in: query
+      name: portVisitReference
+      description: |
+        The **Port Visit Reference** to filter by. The result will only return **Port Call Service** objects including the `portVisitReference`.
```

```suggestion
        The **Port Visit Reference** to filter by. The response will only return objects having the supplied value.
```
Not sure if you like this change, but if you do, you can apply this to all below fields and it is shorter, easier and the same. Also not duplicating the field name. Just my thoughts.

---

I will wait with modifying the filter descriptions as it has not yet been agreed if it should be mandatory to include the filter value in each of the response objects...